src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -133,13 +136,35 @@ void OTARequestor::OnQueryImageResponse(void * context, const QueryImageResponse
             return;
         }
 
-        MutableByteSpan updateToken(requestorCore->mUpdateTokenBuffer);
-        CopySpanToMutableSpan(update.updateToken, updateToken);
-        requestorCore->mUpdateVersion = update.softwareVersion;
-        requestorCore->mUpdateToken   = updateToken;
+        if (response.softwareVersion.HasValue())
```

Comment:
It's already checked in `ExtractUpdateDescription` as it shall be present if Update Available status has been returned. You can refer to `update.softwareVersion` directly at this point.